liquibaseCatalogName not include in generated SQL

**Liquibase Snowflake version** : 4.4.3
**Database Vendor & Version**: snowflake

## Description
The calalog name is not include in generated sql orders for liquibase objects (with snowflake database)
For snowflake, catalogName represents database name

So if the databasechangelog tables are not in the same database with another schema name, Liquibase returns that the schema does not exists

## Steps To Reproduce
Set liquibaseCatalogName and liquibaseSchemaName in properties file :
liquibaseCatalogName: MY_DB
liquibaseSchemaName: MY_SCHEMA
Run liquibase updateSQL

## Actual Behavior
generateed SQL orders :

CREATE TABLE **MY_SCHEMA**.DATABASECHANGELOGLOCK (ID INT NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TIMESTAMP_NTZ, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID));

## Expected/Desired Behavior
generated SQL order should be

CREATE TABLE **MY_DB**.**MY_SCHEMA**.DATABASECHANGELOGLOCK (ID INT NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TIMESTAMP_NTZ, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID));



┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-2039) by [Unito](https://www.unito.io)

When I provided all 4 values ( db, schema, warehouse, and role) in the URL, it worked. If the value for the role is not provided, then Liquibase throws a generic error 
"Caused by: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Error executing SQL SELECT COUNT(*) FROM DATABASECHANGELOGLOCK: Cannot perform SELECT. This session does not have a current database. Call 'USE DATABASE', or use a qualified name."
So, this issue is resolved for me as of now ( I am not worried about the wrong error message :)

--- a/liquibase-integration-tests/src/test/java/liquibase/statementexecute/AddUniqueConstraintExecutorTest.java
+++ b/liquibase-integration-tests/src/test/java/liquibase/statementexecute/AddUniqueConstraintExecutorTest.java
@@ -160,7 +160,7 @@ public class AddUniqueConstraintExecutorTest extends AbstractExecuteTest {
         assertCorrect("alter table \"lbcat2\".[adduqtest] add constraint [uq_test] unique ([coltomakeuq])", DB2Database.class, Db2zDatabase.class);
         assertCorrect("alter table [lbschem2].[adduqtest] add constraint [uq_test] unique ([coltomakeuq])", H2Database.class);
         assertCorrect("alter table [lbschem2].[adduqtest] add constraint [uq_test] unique ([coltomakeuq])", Ingres9Database.class);
-        assertCorrect("alter table [lbschem2].[adduqtest] add constraint [uq_test] unique ([coltomakeuq])", SnowflakeDatabase.class);
+        assertCorrect("alter table [lbcat2].[lbschem2].[adduqtest] add constraint [uq_test] unique ([coltomakeuq])", SnowflakeDatabase.class);
         assertCorrectOnRest("alter table [lbcat2].[adduqtest] add constraint [uq_test] unique ([coltomakeuq])");
 
     }

--- a/liquibase-snowflake/src/main/java/liquibase/database/core/SnowflakeDatabase.java
+++ b/liquibase-snowflake/src/main/java/liquibase/database/core/SnowflakeDatabase.java
@@ -9,6 +9,9 @@ import liquibase.exception.DatabaseException;
 import liquibase.executor.ExecutorService;
 import liquibase.statement.core.RawSqlStatement;
 import liquibase.structure.DatabaseObject;
+import liquibase.structure.core.StoredProcedure;
+import liquibase.structure.core.Table;
+import liquibase.structure.core.View;
 import liquibase.util.SystemUtil;
 
 import java.math.BigInteger;
@@ -102,7 +105,7 @@ public class SnowflakeDatabase extends AbstractJdbcDatabase {
 
     @Override
     public boolean supportsCatalogInObjectName(Class<? extends DatabaseObject> type) {
-        return false;
+        return type == Table.class || type == View.class || type == StoredProcedure.class;
     }
 
     @Override

